### PR TITLE
Add the ability to pararellize Scenario Outlines

### PR DESCRIPTION
**Pull request by vsiakka**
@midniteio @guykisel @efokschaner

### CODE REVIEW

File: src/lib/feature-finder.js [hunk 1]
```diff
@@ -22,10 +22,27 @@ export default function(cucumberOptions) {
       let featureData = parseFeature(file);
```
Comment (vsiakka) [inline src/lib/feature-finder.js:22]: This is how a parsed feature with a scenario outline with 2 examples looks like:

```
{
  "type": "Feature",
  "tags": [],
  "location": {
    "line": 1,
    "column": 1
  },
  "language": "en",
  "keyword": "Feature",
  "name": "X",
  "children": [
    {
      "type": "ScenarioOutline",
      "tags": [
      ],
      "location": {
        "line": 7,
        "column": 3
      },
      "keyword": "Scenario Outline",
      "name": "I am an outline",
      "steps": [
        {
          "type": "Step",
          "location": {
            "line": 50,
            "column": 3
          },
          "keyword": "Then ",
          "text": "xxxx"
        }
      ],
      "examples": [
        {
          "type": "Examples",
          "tags": [],
          "location": {
            "line": 52,
            "column": 5
          },
          "tableBody": [
            {
              "type": "TableRow",
              "location": {
                "line": 54,
                "column": 7
              },
              "cells": [
                {
                  "type": "TableCell",
                  "location": {
                    "line": 54,
                    "column": 13
                  },
                  "value": "x"
                }
              ]
            },
            {
              "type": "TableRow",
              "location": {
                "line": 55,
                "column": 7
              },
              "cells": [
                {
                  "type": "TableCell",
                  "location": {
                    "line": 55,
                    "column": 13
                  },
                  "value": "x"
                }
              ]
            }
          ]
        }
      ]
    }
  ]
}
```

**Comment by guykisel** [comment]
👍

File: src/lib/feature-finder.js [hunk 2]
```diff
@@ -22,10 +22,26 @@ export default function(cucumberOptions) {
...
         .filter((child) => {
-          return (child.type === "Scenario" || child.type === "ScenarioOutline")
+          return (child.type === 'Scenario' || child.type === 'ScenarioOutline')
                  && verifyTags(child, cucumberOptions.tags);
         })
         .map((scenario) => {
+/*eslint-disable max-nested-callbacks*/
+          if (scenario.type === 'ScenarioOutline') {
+            return scenario.examples.map((example) => {
+              return example.tableBody.filter((tableBody) => {
```
Comment (midniteio) [inline src/lib/feature-finder.js]: Single line example: filter(tableBody => tableBody.type === 'TableRow')

File: src/lib/feature-finder.js [hunk 3]
```diff
@@ -22,10 +22,24 @@ export default function(cucumberOptions) {
       let featureData = parseFeature(file);
       return featureData.children
         .filter((child) => {
-          return (child.type === "Scenario" || child.type === "ScenarioOutline")
+          return (child.type === 'Scenario' || child.type === 'ScenarioOutline')
                  && verifyTags(child, cucumberOptions.tags);
         })
         .map((scenario) => {
+/*eslint-disable max-nested-callbacks*/
```
Comment (efokschaner) [inline src/lib/feature-finder.js:29]: Some of this could probably be extracted to a named function. Might improve clarity.

**Pull request merged**